### ISSUE TITLE
Dropout not used correctly in machine_translation example

### ISSUE THREAD

**Issue by aalmah** [mila-udem/blocks-examples#46]
In this example, dropout is effectively ignored because:

- The cost used in GradientDescent should be cg.outputs[0] rather than 'cost':
https://github.com/mila-udem/blocks-examples/blob/12013e6c737171ce285b27381f0f5aacfdff7cab/machine_translation/__init__.py#L160

- Also the cost used in TrainingDataMonitoring:
https://github.com/mila-udem/blocks-examples/blob/12013e6c737171ce285b27381f0f5aacfdff7cab/machine_translation/__init__.py#L112
is not the dropped out one, so instead one should use here cg.outputs[0].

**Comment by dwf** [comment]
@orhanf thoughts on this?

**Comment by orhanf** [comment]
The first item should definitely be fixed, but for the second one, there are some differences between implementations. What we usually monitor is the bleu score or log-prob of the validation set or a subset of the training set. And for both, we use models without any regularization. So, I think, the first item should be fixed, the second can be optional.

**Comment by papar22** [comment]
in which part of the code, do you monitor log-prob of the validation set?
